base2.go

```diff
@@ -0,0 +1,56 @@
+package multibase
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// binaryEncodeToString takes an array of bytes and returns
+// multibase binary representation
+func binaryEncodeToString(src []byte) string {
+	dst := make([]byte, len(src)*8)
+	encodeBinary(dst, src)
+	return string(dst)
+}
+
+// encodeBinary takes the src and dst bytes and converts each
+// byte to their binary rep using power reduction method
+func encodeBinary(dst []byte, src []byte) {
+	for i := 0; i < len(src); i++ {
+		t := src[i]
+		for j := i << 3; j < (i<<3)+8; j++ {
+			higherPower := math.Pow(2, float64(7-(j&7)))
```

Comment:
Untested but, this is what I mean by bit shifting:

```go
for i, b := range src {        // for each byte
	for j := 0; j < 8; j++ {   // for each bit
		if b&(1<<(7-j)) == 0 { // if the bit is 0
			dst[i*8 + j] = '0' // store '0'
		} else {               // otherwise
			dst[i*8 + j] = '1' // store '1'
		}
	}
}
```

Reply:
This looks like a more performant solution.